requirements-tests.txt

```diff
@@ -21,7 +21,7 @@ stubdefaulter==0.1.0
 termcolor>=2.3
 tomli==2.0.1
 tomlkit==0.11.8
-typing-extensions
+typing_extensions @ git+https://github.com/python/typing_extensions.git
```

Comment:
Yes, but that only fixes the problem right now. I think long-term it's good to reflect typing-extensions changes in the stub as soon as they are merged, so that type checkers have more time to pick them up before a release.

Reply:
The pytype crash is presumably because pytype has its own version of `typing.pyi` (https://github.com/google/pytype/blob/main/pytype/stubs/builtins/typing.pytd), and our stub `typing_extensions` is probably re-exporting things from `typing` that pytype doesn't have in its version of `typing`.

If this is going to happen every time a new typing_extensions feature release is cut, then I'm not sure we should be using the typing_extensions `main` branch in our CI; I feel like it would be better to pin typing_extensions and manually update it when a release is cut.

Reply:
Installing the typing_extensions `main` branch in CI now causes our mypy-py37 job to crash, because the `typing_extensions` main branch uses PEP_570 syntax that can't be parsed on Python 3.7.

We have to run our mypy-py37 CI job _on_ Python 3.7 (rather than just with the `--python-version=3.7` flag), or pip will install a version of numpy in CI that's incompatible with Python 3.7, which will cause mypy to fail on e.g. our `tensorflow` stubs.

Reply:
Even if we worked around the numpy issue for `mypy_test.py`, stubtest stdlib is still going to crash on py37 if we install the typing_extensions `main` branch in CI, and that's an even harder problem to solve

Reply:
```suggestion
typing_extensions
```